### PR TITLE
fix: revert deletion of tests auto-run after pull_request changes

### LINKED ISSUE #19: [GitHub Project] auto-trigger tests run on CI before merging changes to the base branch (`main`)
## ✏️ Description

Auto-trigger test-runner job on Github Actions CI before merging changes to the base branch (main).

## 🔧 Why We Need This

We will ensure that our changes do not result in breaking tests.

## 🚀 Proposed Solution
as per description in jaktestowac course:
https://jaktestowac.pl/course/ciagla-integracja-i-testy-automatyczne-z-playwright-i-github-actions/

## ✅ Issue reporting checklist

- [x] I created a descriptive title for this issue
- [x] I checked that it hasn't been reported yet
- [x] I assigned the issue or added `help-wanted` label
- [x] I added appropriate labels

### PR DESCRIPTION
**Pull request by bugITwhisperer**
## Continuation of PR #63
which was merged by accident

### Scope of changes 
- ~revert deletion of running on pr changes~ (must have been fixed in another PR, comment by @kat-kan)
- skip running tests if the changes in PR include only documentation changes (only `.md` files are modified)
- add CI badge w/ current main testing branch status
- remove running tests on every push to tests branch (to be discussed; it was initially a part of PR but can be reverted, comment by @kat-kan)

### PR steps completed:
- [x] The PR is associated with the relevant ISSUE
- [x] all tests have PASS status
- [x] Commit messages follow the `kawqa-gad-playwright` project's [git commit message format](https://github.com/kat-kan/kawqa-gad-playwright/CONTRIBUTING.md)

### CODE REVIEW

**Comment by kat-kan** [comment]
It's not visible in PR #63 yet somehow most of the tests are currently skipped on the main branch. Is it something that is necessary?

**Review by kat-kan — APPROVED** [review]

**Review by adamcegielka — APPROVED** [review]

**Comment by bugITwhisperer** [comment]
@kat-kan : they should not be skipped -  I've just removed the `skip` condition from the test files

**Comment by bugITwhisperer** [comment]
@kat-kan , @adamcegielka , @KarolinaZakrzewska , @Justyna-KO : could you please review the changes? 

Also, please note that I marked 3 tests as fixme from the `tests\api\articles-post.spec.ts suite`:
- Returns 401 Unauthorized after sending article JSON with missing userId
- Returns 422 Unprocessable content after sending article JSON with too long value
- Returns 201 Created after creating article

**Review by Justyna-KO — DISMISSED** [review]

**Review by adamcegielka — DISMISSED** [review]

**Comment by kat-kan** [comment]
@Justyna-KO @Slawomir-DKl I updated this branch and PR description, the history of branch is overwhelming but eventually the changes not so much :) please review!

**Review by Slawomir-DKl — COMMENTED** [review]
I have one remark only

File: .github/workflows/pw-tests.yml [hunk 1]
```diff
@@ -1,9 +1,11 @@
 name: Playwright API Auto-Checks
 on:
   push:
-    branches: ['main', 'tests/ISSUE*']
+    branches: ['main']
```
[the remark this answers is not in the document]
Comment (kat-kan) [reply .github/workflows/pw-tests.yml:4]: Of course , we can modify our test running strategy. We can even run them all on every push to repository no matter which branch.

The only limitation is monthly usage on GH Actions which is counted for my private account. Monthly limit is 2 000 minutes. Right now this is not a problem as 1) our tests take very little time 2) we are not very active here ;)
But it is possible that some time in the future we will have to restrain to some branches only.

[the remark this answers is not in the document]
Comment (Slawomir-DKl) [reply .github/workflows/pw-tests.yml:4]: @kat-kan Now I understand the reason and I want to revert my comment :) - I think there is no need to push your GH limit to its boundaries, especially if you would like to set up some of your own projects. So I resolve the conversation and approve this PR. Thank you

**Review by Slawomir-DKl — APPROVED** [review]

**Comment by kat-kan** [comment]
I'll create new PR for more clarity.

**Pull request closed without merging**